prow/splice/main.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println(dir)
+	s := &splicer{dir}
+	err = s.gitCalls([][]string{
+		{"init"},
+		{"config", "--local", "user.name", "K8S Prow Splice"},
+		{"config", "--local", "user.email", "splice@localhost"},
+	})
+	if err != nil {
+		s.cleanup()
+		return nil, err
+	}
+	return s, nil
+}
+
+// cleanup recurisvely deletes the repository
+func (s *splicer) cleanup() {
+	os.RemoveAll(s.dir)
+}
+
+// gitCall is a helper to call `git -C $path $args`.
+func (s *splicer) gitCall(args ...string) error {
+	fullArgs := append([]string{"-C", s.dir}, args...)
+	output, err := call("git", fullArgs...)
+	fmt.Print(output)
+	return err
+}
+
+// gitCalls is a helper to chain repeated gitCall invocations,
+// returning the first failure, or nil if they all succeeded.
+func (s *splicer) gitCalls(argsList [][]string) error {
+	for _, args := range argsList {
+		err := s.gitCall(args...)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// findMergeable fetches given PRs from upstream, merges them locally,
+// and finally returns a list of PRs that can be merged without conflicts.
+func (s *splicer) findMergeable(remote string, prs []int) ([]int, error) {
+	args := []string{"fetch", remote, "master:master"}
+	for _, pr := range prs {
+		args = append(args, fmt.Sprintf("pull/%d/head:pr/%d", pr, pr))
+	}
+
+	err := s.gitCalls([][]string{
+		{"reset", "--hard"},
+		{"checkout", "--orphan", "blank"},
+		{"reset", "--hard"},
+		{"clean", "-fdx"},
+		args,
+		{"checkout", "-B", "batch", "master"},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for i, pr := range prs {
+		err := s.gitCall("merge", "--no-ff", "--no-stat",
+			"-m", fmt.Sprintf("merge #%d", pr),
+			fmt.Sprintf("pr/%d", pr))
+		if err != nil {
+			return prs[:i], nil
+		}
+	}
+	return prs, nil
+}
+
+// gitRef returns the SHA for the given git object-- a branch, generally.
+func (s *splicer) gitRef(ref string) string {
+	output, err := call("git", "-C", s.dir, "rev-parse", ref)
+	if err != nil {
+		return ""
+	}
+	return strings.TrimSpace(output)
+}
+
+type prRef struct {
+	PR  int    `json:"pr"`
+	SHA string `json:"sha"`
+}
+
+type buildRef struct {
+	Repo string  `json:"repo"`
+	Base string  `json:"base_ref"`
+	PRs  []prRef `json:"prs"`
+}
+
+// Produce a struct describing the exact revisions that were merged together.
+// This is sent to Prow and used by the Submit Queue when deciding whether it
+// can use a batch test result to merge something.
+func (s *splicer) makeBuildRef(repo string, prs []int) *buildRef {
+	ref := &buildRef{Repo: repo, Base: s.gitRef("master")}
+	for _, pr := range prs {
+		branch := fmt.Sprintf("pr/%d", pr)
+		ref.PRs = append(ref.PRs, prRef{pr, s.gitRef(branch)})
+	}
+	return ref
+}
+
+func main() {
+	flag.Parse()
+
+	repo := githubRepoRE.FindString(*remoteURL)
+
+	splicer, err := makeSplicer()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer splicer.cleanup()
+
+	// Loop endless, sleeping a minute between iterations
+	for {
```

Comment:
`for range time.Tick(1*time.Minute) {` so that you don't need the sleeps scattered throughout.